pkg/endpoint/directory.go

```diff
@@ -48,27 +48,43 @@ func (e *Endpoint) backupDirectoryPath() string {
 
 // moveNewFilesTo copies all files, that do not exist in newDir, from oldDir.
 func moveNewFilesTo(oldDir, newDir string) error {
-	oldFiles, err := os.ReadDir(oldDir)
+	var err error
+
+	oldDirFile, err := os.Open(oldDir)
+	if err != nil {
+		return err
+	}
+	defer oldDirFile.Close()
+
+	oldFiles, err := oldDirFile.Readdirnames(-1)
 	if err != nil {
 		return err
 	}
-	newFiles, err := os.ReadDir(newDir)
+
+	newDirFile, err := os.Open(newDir)
 	if err != nil {
 		return err
 	}
+	defer newDirFile.Close()
+
+	newFiles, err := newDirFile.Readdirnames(-1)
+	if err != nil {
+		return err
+	}
+
+	newFilesHash := make(map[string]bool, len(newFiles))
+	for _, f := range newFiles {
+		newFilesHash[f] = true
+	}
```

Comment:
`newFilesHash` can use `struct{}` as a value since `true` is the only value that's ever set in the map.

Reply:
Ah good catch, changed.